### ISSUE TITLE
Multipart upload uploads a file with size of 5MB and not original file size

### ISSUE THREAD

**Issue by lonormaly** [aws/aws-sdk-js#778]
Using Multipart upload, the uploaded file is eventually only 5MB and not the whole size.

I use leavePartsOnError so resuming is enabled, if it helps in figuring it out.
The frequency is higher than it was in previous version of the SDK (again, not sure if it's related) and stands on about 40% of downloads (!!!).
I use Cognito authentication.

![image](https://cloud.githubusercontent.com/assets/3742919/10966299/576de980-83bb-11e5-9eee-46fa6096e06d.png)

**Comment by chrisradek** [comment]
@lonormaly 
Can you give some details on what is causing the initial multipart upload to abort? 

One thing that might be useful is to turn on the logger for S3 while you're troubleshooting (you can set the `logger` parameter to `console` when initializing S3 in the SDK).

Using the logger, you can verify that all the `uploadPart` operations for a given upload have the same `UploadId` and that you're seeing the expected number of `uploadPart` operations.  Also verify that the `completeMultipartUpload` command shows the number of parts you'd expect.

Let me know if this helps and what you find!

**Comment by lonormaly** [comment]
The http progress announces "total" of 5MB and part 1 instead of total of original file size.
It happened to me especially when disconnecting from internet and reconnecting again for some reason.

**Comment by kentor** [comment]
Yes in playing around the resuming feature, I can reproduce this using Buffers

script (I'm using node 4.2)

```js
const AWS = require('aws-sdk');

const credentials = new AWS.Credentials({
  accessKeyId: ...,
  secretAccessKey: ...,
});

const s3 = new AWS.S3({
  credentials,
  params: { Bucket: ... },
  region: ...,
});

const MB = 1024 * 1024;
var body = new Buffer(20 * MB);
body.fill('1', 0, 5 * MB);
body.fill('2', 5 * MB, 10 * MB);
body.fill('3', 10 * MB, 15 * MB);
body.fill('4', 15 * MB);

const upload = new AWS.S3.ManagedUpload({
  leavePartsOnError: true,
  params: { Body: body, Key: 'test' },
  service: s3,
});

upload.on('httpUploadProgress', function(progress) {
  console.log(progress);
});

upload.send = upload.send.bind(upload, (err, data) => {
  if (err) {
    console.log('error:', err);
  } else {
    console.log('success:', data);
  }
});

upload.send();

setTimeout(function() {
  console.log('abort!');
  upload.abort();
  setTimeout(function() {
    upload.send();
  }, 2000);
}, 2000);
```

**Comment by lonormaly** [comment]
Thanks, but we don't use streams and yet we still get this error quite a bit

**Comment by chrisradek** [comment]
@kentor @lonormaly 
Thanks for the code you've provided to reproduce the issue. I plan on taking a look at this issue and the pull requests as well shortly.

**Comment by davidmaxwaterman** [comment]
Is this issue fixed? Almost 2 years old...and I'm having trouble getting 'resume' functionality to work.

**Comment by jeskew** [comment]
Fixed in #1550
